server/channels/app/post.go

```diff
@@ -774,109 +765,108 @@ func (a *App) UpdatePost(c request.CTX, receivedUpdatedPost *model.Post, safeUpd
+			// In this case, it will broadcast the message with metadata wiped out
+			return nil
 		}
-		return false, err
-	}
-
-	userIDs, nErr := a.Srv().Store().Channel().GetAllChannelMemberIdsByChannelId(post.ChannelId)
-	if nErr != nil {
-		a.CountNotificationReason(model.NotificationStatusError, model.NotificationTypeAll, model.NotificationReasonFetchError, model.NotificationNoPlatform)
-		a.NotificationsLog().Error("Cannot get channel members",
-			mlog.String("type", model.NotificationTypeWebsocket),
-			mlog.String("post_id", post.Id),
-			mlog.String("status", model.NotificationStatusError),
-			mlog.String("reason", model.NotificationReasonFetchError),
-			mlog.String("referenced_post_id", previewedPostID),
-			mlog.Err(nErr),
-		)
-		return false, model.NewAppError("publishWebsocketEventForPermalinkPost", "app.channel.get_members.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		return appErr
 	}
 
-	permalinkPreviewedChannel, err := a.GetChannel(c, previewedPost.ChannelId)
-	if err != nil {
-		if err.StatusCode == http.StatusNotFound {
+	permalinkPreviewedChannel, appErr := a.GetChannel(c, previewedPost.ChannelId)
+	if appErr != nil {
+		if appErr.StatusCode == http.StatusNotFound {
 			a.CountNotificationReason(model.NotificationStatusError, model.NotificationTypeAll, model.NotificationReasonFetchError, model.NotificationNoPlatform)
 			a.NotificationsLog().Error("Cannot get channel",
 				mlog.String("type", model.NotificationTypeWebsocket),
 				mlog.String("post_id", post.Id),
 				mlog.String("status", model.NotificationStatusError),
 				mlog.String("reason", model.NotificationReasonFetchError),
-				mlog.String("referenced_post_id", previewedPostID),
+				mlog.String("referenced_post_id", previewedPost.Id),
 			)
 			c.Logger().Warn("channel containing permalinked post not found", mlog.String("referenced_channel_id", previewedPost.ChannelId))
-			return false, nil
-		}
-		return false, err
-	}
-
-	originalEmbeds := post.Metadata.Embeds
-	originalProps := post.GetProps()
-	permalinkPreviewedPost := post.GetPreviewPost()
-	for _, userID := range userIDs {
-		if permalinkPreviewedPost != nil {
-			post.Metadata.Embeds = originalEmbeds
-			post.SetProps(originalProps)
-		}
-
-		postForUser := a.sanitizePostMetadataForUserAndChannel(c, post, permalinkPreviewedPost, permalinkPreviewedChannel, userID)
-
-		// Using DeepCopy here to avoid a race condition
-		// between publishing the event and setting the "post" data value below.
-		messageCopy := message.DeepCopy()
-		broadcastCopy := messageCopy.GetBroadcast()
-		broadcastCopy.UserId = userID
-		messageCopy.SetBroadcast(broadcastCopy)
-
-		postJSON, jsonErr := postForUser.ToJSON()
-		if jsonErr != nil {
-			c.Logger().Warn("Failed to encode post to JSON", mlog.Err(jsonErr))
+			// In this case, it will broadcast the message with metadata wiped out
+			return nil
 		}
-		messageCopy.Add("post", postJSON)
-		a.Publish(messageCopy)
+		return appErr
```

Comment:
This is a scenario where it feels like we still ought to publish, just without the permalink preview.

Reply:
> If we publish without the permalink preview in this case, the same reasoning would also apply for a.GetSinglePost?

Yes, I'd propose doing that as well.

For me, resolving the permalink preview is an optional value-add -- users without access already have to do without it. It seems strictly worse to effectively create a missed message.

Reply:
It sounds like we are already partially handling errors (not found case) so what would be the issue in handling them all the same way?

Reply:
The problem is that - in that case, appErr will no longer be logged. And if you have to log it, then you end up with multiple warn log lines throughout the code. To properly do it, you would need to differentiate between "errors that should just be logged, but code can move ahead" vs "errors which should be logged and code should return". All of this is additional overhead to me. I'd rather fail clearly, rather than fail partially and try to do a half job.

Reply:
Could we use `setupBroadcastHookForPermalink` as the boundary? Always return errors from this method (even for `!model.IsValidId(previewProp)`, but just log the error and universally adopt the semantics `In this case, it will broadcast the message with metadata wiped out`.

Reply:
I could, but if `post.ToJSON()` returns an error, this could lead to a situation where the original post might have the metadata stripped out, but still gets broadcasted with the full unsanitized payload. I don't think that would be intentional.

While I agree that this is a good exercise, but I feel this is starting to go out of scope. I am pretty sure there are lots of places in the codebase where there is inconsistent behavior. I'd suggest that we perhaps revisit this problem at a global scope and come up with some best practices for everyone to follow.